Source/WebCore/html/HTMLTableElement.cpp

```diff
@@ -81,9 +81,9 @@ ExceptionOr<void> HTMLTableElement::setCaption(RefPtr<HTMLTableCaptionElement>&&
 
 RefPtr<HTMLTableSectionElement> HTMLTableElement::tHead() const
 {
-    for (RefPtr<Node> child = firstChild(); child; child = child->nextSibling()) {
+    for (Ref child : childrenOfType<HTMLTableSectionElement>(const_cast<HTMLTableElement&>(*this))) {
         if (child->hasTagName(theadTag))
-            return downcast<HTMLTableSectionElement>(child.get());
+            return child;
```

Comment:
I wonder whether using `WTFMove` would be helpful, irrelevant, or harmful. We do want to hand off ownership, not churn the reference count, is possible,

Reply:
@cdumez could you help with this?

Reply:
Like Darin, I'm uncertain here what's best, it would require testing. You might want to try but I have a vague recollection that gcc (and thus glib EWS) will complain about a redundant WTFMove() if you try to use it in such cases. If that's the case, it would seem to indicate that the WTFMove() is not needed.

Reply:
No compiler issue on macOS with adding WTFMove. Pushed it to the bots now.

Reply:
gcc doesn't like it though:
```
In file included from /app/webkit/WebKitBuild/WPE/Release/WTF/Headers/wtf/FastMalloc.h:26,
                 from /app/webkit/Source/WebCore/config.h:47,
                 from /app/webkit/Source/WebCore/html/HTMLTableCaptionElement.cpp:25,
                 from /app/webkit/WebKitBuild/WPE/Release/WebCore/DerivedSources/unified-sources/UnifiedSource-950a39b6-14.cpp:1:
/app/webkit/Source/WebCore/html/HTMLTableElement.cpp: In member function ‘WTF::RefPtr<WebCore::HTMLTableSectionElement> WebCore::HTMLTableElement::tHead() const’:
/app/webkit/WebKitBuild/WPE/Release/WTF/Headers/wtf/StdLibExtras.h:675:58: error: redundant move in return statement [-Werror=redundant-move]
  675 | #define WTFMove(value) std::move<WTF::CheckMoveParameter>(value)
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
/app/webkit/Source/WebCore/html/HTMLTableElement.cpp:86:20: note: in expansion of macro ‘WTFMove’
   86 |             return WTFMove(child);
```

Reply:
So my memory was correct and the original code was correct.

Reply:
Indeed. Will revert that change and land the prior patch then.